sound/soc/sof/ipc4-topology.c

```diff
@@ -1604,16 +1599,15 @@ static int sof_ipc4_init_input_audio_fmt(struct snd_sof_dev *sdev,
 		type = sof_ipc4_fmt_cfg_to_type(fmt->fmt_cfg);
 		if (params_rate(params) == rate && params_channels(params) == channels &&
 		    sample_valid_bits == valid_bits && sample_type == type)
-			break;
+			goto in_fmt;
```

Comment:
stylistically I think the commonly used pattern, where `goto` is *only* used for error handling is clearer. IIUC this change is using `goto` to jump to a success case continuation, while handling the error on the main code path. Also it doesn't seem to be related to the fix. The actual fix seems to be removing the lines 1579-1582 (and removing the jump label), that would also make the patch smaller